### ISSUE TITLE
variable prompt is ignoring the target option

### ISSUE THREAD

**Issue by Madhu-mohan86** [hashicorp/terraform#34878]
### Terraform Version

```shell
[crazyphil@archlinux ~]$ terraform --version
Terraform v1.7.5
on linux_amd64
```

### Terraform Configuration Files

```
variable "data3" {
  type = string
 default="example value for the .tfvars file functionality check "
}

resource "local_file" "version4" {
  content = var.data3
  filename = "tfvarsexample"
}
```

### Debug Output

https://github.com/Madhu-mohan86/terraform_issue_debugfiles.git

### Expected Behavior

The issue is when i used ```-target ``` flag to provision my specific resource that is **version4** but variable prompt for resource **version2** is displaying in cli which is the proper behaviour for ```-target``` flag I guess 
```
[madhu@archlinux local]$ terraform plan -var-file=var.tfvars -target=local_file.version4
var.data2
  Enter a value:
```
you can clearly see that i have no variable defined in my configuration i had only variable defined with  data3

### Actual Behavior

the actual output for the command should be the prompt should not be displayed which is irrelevant for ```-target``` flag

### Steps to Reproduce

```terraform init```
```terraform plan ```

### Additional Context

as i described my issue in the expected and actual behaviour section once i again wrap it up with minimal the issue is variable input prompt intended another resource is conflicting with plan of the target resource so the issue is simple target flag is just not checking the variable prompt is intended for the given resource or another ignorable resource

**Comment by crw** [comment]
If you are viewing this issue and would like to indicate your interest, please use the 👍 reaction on the issue description to upvote this issue. We also welcome additional use case descriptions. Thanks!

**Comment by Madhu-mohan86** [comment]
Yeah ofcourse I can give the use cases so this might not be in production but for testing or debugging environments it may help , so in this case if some data needed to be default or entitled to a specific resource and assume it causes some error or misconfiguration so to debug that I can have ```-target``` flag as my one possibility but since target is still not scoped for that source this activity from another may cause some another set of hurdle to debug it . My only context is if the ```target ``` flag is made like full fledged isolated environment  probably debugging the configuration will be lot easier
